pkg/bundle/sync.go

```diff
@@ -777,3 +778,38 @@ func (b *bundle) migrateConfigMapToApply(ctx context.Context, obj client.Object,
 	cm.SetManagedFields(managedFields)
 	return true, b.directClient.Update(ctx, &cm)
 }
+
+// remove duplicate certificates from bundles
+func deduplicateBundles(bundles []string) []string {
+	var block *pem.Block
+
+	var certificatesHashes = make(map[[32]byte]struct{})
+	var dedupCerts []string
+
+	for _, cert := range bundles {
+		certBytes := []byte(cert)
+	LOOP:
+		for {
+			block, certBytes = pem.Decode([]byte(certBytes))
+			if block == nil {
+				break LOOP
```

Comment:
Error return parameter instead?

Reply:
No, it this case it's not error, it's just indicates no more data in this particular source, however there are might be the next source with data but instead process that data we throw error.
However error return was added here:

```
if block.Type != "CERTIFICATE" {
      return nil, fmt.Errorf("couldn't decode PEM block containing certificate")
}
```